pkgs/applications/accessibility/nerd-dictation/default.nix

```diff
@@ -0,0 +1,51 @@
+{ fetchFromGitHub
+, substituteAll
+, lib
+, stdenv
+, python3
+, pulseaudio
+, xdotool
+, ydotool
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "nerd-dictation";
+  version = "unstable-2022-07-12";
+
+  src = fetchFromGitHub {
+    owner = "ideasman42";
+    repo = "nerd-dictation";
+    rev = "6194641846e55b89c955a0fd6b3ac83bfd52b7d7";
+    hash = "sha256-fvdh2eZHh4Ups85BNejedo/tSf3cb/C7pJvgf9zKqoU=";
+  };
+
+  patches = [
+    (substituteAll {
+      src = ./fix-paths.patch;
+      parec = "${pulseaudio}/bin/parec";
+      xdotool = "${xdotool}/bin/xdotool";
+      ydotool = "${ydotool}/bin/ydotool";
+    })
+  ];
+
+  propagatedBuildInputs = with python3.pkgs; [
+    vosk-python
+  ];
+
+  postPatch = ''
+    cd package/python
+  '';
+
+  # No tests.
+  doCheck = false;
+
+  # Broken for some reason.
```

Comment:
Can you paste the log?

Reply:
```
pythonCatchConflictsPhase
Found duplicated packages in closure for dependency 'nerd-dictation': 
  nerd-dictation 0.0.0 (/nix/store/3dc8q0l76p27nnyv8b8aijfhs6ssifk7-nerd-dictation-unstable-2022-07-12/lib/python3.10/site-packages)
  nerd-dictation 0.0.0 (/nix/store/3dc8q0l76p27nnyv8b8aijfhs6ssifk7-nerd-dictation-unstable-2022-07-12/lib/python3.10/site-packages)

Package duplicates found in closure, see above. Usually this happens if two packages depend on different version of the same dependency.
```

Reply:
> nerd_dictation-0.0.0.dist-info

The version number is not set correct.

Reply:
https://github.com/ideasman42/nerd-dictation/blob/master/package/python/setup.py#L43

The version number is hardcoded to ``0.0.0`` which is the default value and should be treated as an error.

Reply:
The layout in site-packages is really strange and very custom.
Anything installed under site-packages should be importable and it is frowned upon to files directly under site-packages.

I expect this to cause more trouble in the future.

Compared to a standard python package:

```
 ➜ tree result
 result ⇒ /nix/store/3qfd3gwpar83k89wdnw8pab7xr3bfv52-nerd-dictation-unstable-2022-07-12
├──  bin
│   └──  nerd-dictation
├──  lib
│   └──  python3.10
│       └──  site-packages
│           ├──  __pycache__
│           │   └──  setup.cpython-310.pyc
│           ├──  doc
│           │   └──  readme.rst
│           ├──  nerd-dictation
│           ├──  nerd_dictation-0.0.0.dist-info
│           │   ├──  direct_url.json
│           │   ├──  INSTALLER
│           │   ├──  METADATA
│           │   ├──  RECORD
│           │   ├──  REQUESTED
│           │   ├──  top_level.txt
│           │   └──  WHEEL
│           ├──  nerd_dictation.egg-info
│           │   ├──  dependency_links.txt
│           │   ├──  PKG-INFO
│           │   ├──  requires.txt
│           │   ├──  SOURCES.txt
│           │   └──  top_level.txt
│           ├──  nix_run_setup
│           └──  setup.py
├──  nerd-dictation
│   ├──  nerd-dictation
│   └──  readme.rst
└──  nix-support
    └──  propagated-build-inputs
```

```
 ➜ tree result
 result ⇒ /nix/store/k38smakhffi54jgpj21d4l51w5va85ly-python3.10-jsonschema-4.7.2
├──  bin
│   └──  jsonschema
├──  lib
│   └──  python3.10
│       └──  site-packages
│           ├──  jsonschema
│           │   ├──  __init__.py
│           │   ├──  __main__.py
│           │   ├──  __pycache__
│           │   │   ├──  __init__.cpython-310.pyc
│           │   │   ├──  __main__.cpython-310.pyc
│           │   │   ├──  _format.cpython-310.pyc
│           │   │   ├──  _legacy_validators.cpython-310.pyc
│           │   │   ├──  _reflect.cpython-310.pyc
│           │   │   ├──  _types.cpython-310.pyc
│           │   │   ├──  _utils.cpython-310.pyc
│           │   │   ├──  _validators.cpython-310.pyc
│           │   │   ├──  cli.cpython-310.pyc
│           │   │   ├──  exceptions.cpython-310.pyc
│           │   │   ├──  protocols.cpython-310.pyc
│           │   │   └──  validators.cpython-310.pyc
│           │   ├──  _format.py
│           │   ├──  _legacy_validators.py
│           │   ├──  _reflect.py
│           │   ├──  _types.py
│           │   ├──  _utils.py
│           │   ├──  _validators.py
│           │   ├──  benchmarks
│           │   │   ├──  __init__.py
│           │   │   ├──  __pycache__
│           │   │   │   ├──  __init__.cpython-310.pyc
│           │   │   │   ├──  issue232.cpython-310.pyc
│           │   │   │   └──  json_schema_test_suite.cpython-310.pyc
│           │   │   ├──  issue232
│           │   │   │   └──  issue.json
│           │   │   ├──  issue232.py
│           │   │   └──  json_schema_test_suite.py
│           │   ├──  cli.py
│           │   ├──  exceptions.py
│           │   ├──  protocols.py
│           │   ├──  schemas
│           │   │   ├──  draft2019-09.json
│           │   │   ├──  draft2020-12.json
│           │   │   ├──  draft3.json
│           │   │   ├──  draft4.json
│           │   │   ├──  draft6.json
│           │   │   ├──  draft7.json
│           │   │   └──  vocabularies.json
│           │   ├──  tests
│           │   │   ├──  __init__.py
│           │   │   ├──  __pycache__
│           │   │   │   ├──  __init__.cpython-310.pyc
│           │   │   │   ├──  _helpers.cpython-310.pyc
│           │   │   │   ├──  _suite.cpython-310.pyc
│           │   │   │   ├──  fuzz_validate.cpython-310.pyc
│           │   │   │   ├──  test_cli.cpython-310.pyc
│           │   │   │   ├──  test_deprecations.cpython-310.pyc
│           │   │   │   ├──  test_exceptions.cpython-310.pyc
│           │   │   │   ├──  test_format.cpython-310.pyc
│           │   │   │   ├──  test_jsonschema_test_suite.cpython-310.pyc
│           │   │   │   ├──  test_types.cpython-310.pyc
│           │   │   │   ├──  test_utils.cpython-310.pyc
│           │   │   │   └──  test_validators.cpython-310.pyc
│           │   │   ├──  _helpers.py
│           │   │   ├──  _suite.py
│           │   │   ├──  fuzz_validate.py
│           │   │   ├──  test_cli.py
│           │   │   ├──  test_deprecations.py
│           │   │   ├──  test_exceptions.py
│           │   │   ├──  test_format.py
│           │   │   ├──  test_jsonschema_test_suite.py
│           │   │   ├──  test_types.py
│           │   │   ├──  test_utils.py
│           │   │   └──  test_validators.py
│           │   └──  validators.py
│           └──  jsonschema-4.7.2.dist-info
│               ├──  direct_url.json
│               ├──  entry_points.txt
│               ├──  INSTALLER
│               ├──  licenses
│               │   └──  COPYING
│               ├──  METADATA
│               ├──  RECORD
│               ├──  REQUESTED
│               └──  WHEEL
└──  nix-support
    └──  propagated-build-inputs
```

Reply:
Looks like there is something here: https://github.com/ideasman42/nerd-dictation/pull/79